### PR TITLE
SATs: allow new records in a sequential read for full refresh test

### PR DESCRIPTION
**Pull request by davydov-d**
## What
From time to time the source connector builds fail due to the different output of two subsequent reads. A common case is a new record in the second read which causes a fail. Since it takes time for many connectors to perform a full refresh sync, a new record is very likely to appear in the meantime. 

## How
Allow new records in full refresh test

### CODE REVIEW

**Review by arsenlosenko — APPROVED** [review]

**Comment by davydov-d** [comment]
will publish/merge this one once I get approve from Airbyte please

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_full_refresh.py
Comment (brianjlai) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_full_refresh.py:71]: do we still need to check the symmetric difference if we're effectively doing the validation above by verifying that `records_1` is not a subset of `records_2`?

Thinking about all the permutations, `records_2` should at minimum have the same records as `records_1` but could have more. So I think your above condition is sufficient `set(map(serializer, stream_records_1)).issubset(set(map(serializer, stream_records_2)))`

If we've gotten into this block where we know that `records_1` has records that `records_2` is missing, maybe we should be more descriptive and use `records_1 - records_2` to show which records `records_2` was missing that should have been there. The symmetric difference combines it all together which might be confusing.

**Review by brianjlai — COMMENTED** [review]
let's make sure to test this on a handful more connectors since its a change to the core validation logic and out of curiosity, do you know how often these flaky tests were occurring and which connectors it was most prominent for?

**Comment by alafanechere** [comment]
> Since it takes time for many connectors to perform a full refresh sync, a new record will likely appear in the meantime.

As @brianjlai I'd be interested in knowing for which connector you encounter this problem. As SAT run on our sandbox accounts, most do not have new data very often. But for our GitHub account, it might indeed be the case.

**Comment by davydov-d** [comment]
@brianjlai @alafanechere thanks for the review. I made the appropriate change in the code.
As for the connectors which encountered this problem, I can recall at least source-slack, source-facebook-marketing, source-instagram, source-google-analytics, source-google-ads. I'm sure there were more but that's all I can recall at the moment

File: airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py
Comment (alafanechere) [inline airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py:136]: `test_case_name` is only used for readability isn't it? I suggest leverage pytest params id:
```suggestion
    pytest.params(
        [["id"]],
        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
        [{"id": 1, "first_name": "Albert", "last_name": "Einstein"}, {"id": 2, "first_name": "Joseph", "last_name": "Lagrange"}],
        False,
        id="pk_sets_equal_success",
    ),
```

and not build the id afterward in the list comprehension.

File: airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py
Comment (alafanechere) [inline airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py:137]:
```suggestion
from contextlib import nullcontext as does_not_raise

recordset_comparison_test_cases = [
    (
        [["id"]],
        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
        [{"id": 1, "first_name": "Albert", "last_name": "Einstein"}, {"id": 2, "first_name": "Joseph", "last_name": "Lagrange"}],
        does_not_raise(),
        "pk_sets_equal_success",
    ), 
      (
        [["id"]],
        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
        [{"id": 2, "first_name": "Joseph", "last_name": "Lagrange"}],
        pytest.raises(Failed),
        "pk_sets_equal_success",
    ),
```

And then `if should_fail` is not required.

File: airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py
Comment (alafanechere) [inline airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py:200]: nit
```suggestion
def test_recordset_comparison(mocker, primary_key, first_read_records, second_read_records, should_fail, test_case_name):
```

the `mocker` fixture can be used instead of importing `mock`: `docker_runner_mock = mocker.MagicMock()`

**Review by alafanechere — APPROVED** [review]
I approve to unblock but please implement the suggestions I left on the unit tests.

File: airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py
Comment (davydov-d) [reply airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py:137]: nice solution, thank you!

**Comment by davydov-d** [comment]
> I approve to unblock but please implement the suggestions I left on the unit tests.

thanks, done!

**Comment by davydov-d** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Pull request merged**
